CMakeLists.txt

```diff
@@ -35,6 +35,41 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE RelWithDebInfo)
 endif()
 
+#
+# Address Sanitize
+#
+if(ENABLE_ASAN)
+  #
+  # -fsanitize=address           - Build with address sanitizer support
+  # -fno-omit-frame-pointer      - Always keep the frame pointer in a register
+  # -fno-optimize-sibling-calls  - Don't optimize away tail recursion.
+  #
+  set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=address -fno-omit-frame-pointer -fno-optimize-sibling-calls")
```

Comment:
You can't use multiple -fsanitize flags, as the later values override the previous ones.

You should put the enabled sanitizers in a list, then if the list isn't empty, join the contents with commas, and add it to the CFLAGS and LDFLAGS, we just did this for FreeRADIUS a couple of weeks ago, remember?

i.e. if you had all enabled you'd add `-fanitize=address,leak,undefined`  if you did `-fsanitize=address -fsanitize=leak -fsanitize=undefined` you'd either end up with only the address sanitizer enabled or only ubsan enabled.

Reply:
https://cmake.org/cmake/help/v3.12/command/list.html